test/test_brownian.py

```diff
@@ -31,56 +36,37 @@ def _make_struct(shape, dtype):
 @pytest.mark.parametrize(
     "ctr", [diffrax.UnsafeBrownianPath, diffrax.VirtualBrownianTree]
 )
-@pytest.mark.parametrize(
-    "levy_area", [diffrax.BrownianIncrement, diffrax.SpaceTimeLevyArea]
-)
+@pytest.mark.parametrize("levy_area", levy_areas)
 @pytest.mark.parametrize("use_levy", (False, True))
 def test_shape_and_dtype(ctr, levy_area, use_levy, getkey):
     t0 = 0.0
     t1 = 2.0
 
-    shapes = (
-        (),
-        (0,),
-        (1, 0),
-        (2,),
-        (3, 4),
-        (1, 2, 3, 4),
-        (1, 2, 3, 4),
-        {
-            "a": (1,),
-            "b": (2, 3),
-        },
-        {
-            "a": (1,),
-            "b": (2, 3),
-        },
-        (
-            (1, 2),
-            (
-                (3, 4),
-                (5, 6),
-            ),
-        ),
+    shapes_dtypes = (
+        ((), None),
+        ((0,), None),
+        ((1, 0), None),
+        ((3, 4), jnp.float32),
+        ((1, 2, 3, 4), jnp.float64),
+        ({"a": (1,), "b": (2, 3)}, {"a": None, "b": jnp.float64}),
+        ((2,), jnp.float16),
+        ((1, 2, 3, 4), jnp.complex128),
+        (((1, 2), ((3, 4), (5, 6))), (jnp.float16, (jnp.float32, jnp.float64))),
+        ({"a": (1,), "b": (2, 3)}, {"a": jnp.float64, "b": jnp.complex128}),
     )
 
-    dtypes = (
-        None,
-        None,
-        None,
-        jnp.float16,
-        jnp.float32,
-        jnp.float64,
-        jnp.complex128,
-        {"a": None, "b": jnp.float64},
-        {"a": jnp.float64, "b": jnp.complex128},
-        (jnp.float16, (jnp.float32, jnp.float64)),
-    )
+    if (
+        ctr is diffrax.VirtualBrownianTree
+        and levy_area is diffrax.SpaceTimeTimeLevyArea
+    ):
+        # VBT with STTLA does not support float16 or complex dtypes
+        # because it uses jax.random.multivariate_normal
+        shapes_dtypes = shapes_dtypes[:6]
```

Comment:
This seems a little error prone. Can this be switched into two groups `shapes_dtypes1` and `shapes_dtypes2`, which are then combined or not appropriately?

Reply:
Yes, good point.